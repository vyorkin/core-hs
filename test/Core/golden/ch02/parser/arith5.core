iszero n = n (\x. F) T
plus m n f x = n f (m f x)
mult m n f = m (n f)
pow m n = n m
pred n f x = n (\g h. h (g f)) (\u. x) I
ack = \m. m (\f n.  n f (f one)) succ
sub m n = (n pred) m
