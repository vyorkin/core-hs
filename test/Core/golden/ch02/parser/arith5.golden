[(Name "iszero",[Name "n"],EAp (EAp (EVar (Name "n")) (ELam [Name "x"] (EVar (Name "F")))) (EVar (Name "T"))),(Name "plus",[Name "m",Name "n",Name "f",Name "x"],EAp (EAp (EVar (Name "n")) (EVar (Name "f"))) (EAp (EAp (EVar (Name "m")) (EVar (Name "f"))) (EVar (Name "x")))),(Name "mult",[Name "m",Name "n",Name "f"],EAp (EVar (Name "m")) (EAp (EVar (Name "n")) (EVar (Name "f")))),(Name "pow",[Name "m",Name "n"],EAp (EVar (Name "n")) (EVar (Name "m"))),(Name "pred",[Name "n",Name "f",Name "x"],EAp (EAp (EAp (EVar (Name "n")) (ELam [Name "g",Name "h"] (EAp (EVar (Name "h")) (EAp (EVar (Name "g")) (EVar (Name "f")))))) (ELam [Name "u"] (EVar (Name "x")))) (EVar (Name "I"))),(Name "ack",[],ELam [Name "m"] (EAp (EAp (EVar (Name "m")) (ELam [Name "f",Name "n"] (EAp (EAp (EVar (Name "n")) (EVar (Name "f"))) (EAp (EVar (Name "f")) (EVar (Name "one")))))) (EVar (Name "succ")))),(Name "sub",[Name "m",Name "n"],EAp (EAp (EVar (Name "n")) (EVar (Name "pred"))) (EVar (Name "m")))]